src/modules/todoList.js

```diff
@@ -0,0 +1,142 @@
+          <button class="editList"><i class="fa-solid fa-check"></i></button>
+       </form>
+     `;
+    const editListButton = document.querySelector('.editList');
+    editListButton.addEventListener('click', (event) => {
+      event.preventDefault();
+      const editedValue = document.querySelector('.editInput').value;
+      if (editedValue !== '') {
+        const listCollection = this.#getLists();
+        const listUpdated = listCollection.map((list) => {
+          if (list.index === listIndex) {
+            return {
+              ...list,
+              description: editedValue,
+            };
+          }
+          return list;
+        });
+        this.#setList(listUpdated);
+        this.showList();
+      }
+    });
+  };
+
+  showList() {
+    const todoLists = this.#getLists();
+    if (todoLists.length > 0) {
+      todoLists.sort((a, b) => a.index - b.index);
+      const todoListsWrapper = document.querySelector('.lists');
+      todoListsWrapper.innerHTML = '';
+      todoLists.forEach((list) => {
+        const listWrapper = document.createElement('li');
+        listWrapper.innerHTML = '';
+        listWrapper.classList.add('list');
+        listWrapper.innerHTML = `
+            <div class="content c-${list.index}">
+              <input type="checkbox" id="checkbox-${list.index}" class="checkbox">
+              <p class="description desc-${list.index}">${list.description}</p>
+            </div>
+            <div class="editButton button-${list.index}">
+                <button type="button" class="delete"><i class="fa-solid fa-trash-can"></i></button>  
+            </div>
+              `;
+        todoListsWrapper.appendChild(listWrapper);
+      });
+
+      const checkboxAll = document.querySelectorAll('.checkbox');
+      checkboxAll.forEach((checkBox, index) => {
+        if (todoLists[index].completed) {
+          checkBox.checked = true;
+          const description = document.querySelector(`.desc-${index + 1}`);
+          description.classList.add('completed');
+        }
+      });
+
+      const deleteButton = document.querySelectorAll('.delete');
+      deleteButton.forEach((btn, index) => {
+        btn.addEventListener('click', () => {
+          this.deleteList(index + 1);
+        });
+      });
+
+      const descriptions = document.querySelectorAll('.description');
+      descriptions.forEach((description, index) => {
+        description.addEventListener('click', () => {
+          this.editList(index + 1);
+        });
+      });
+
+      const checkboxButtons = document.querySelectorAll('.checkbox');
+      checkboxButtons.forEach((btn, index) => {
+        btn.addEventListener('click', (event) => {
+          const statusUpdate = new Status();
+          statusUpdate.status(index + 1, event.target.checked);
+        });
+      });
+    } else {
+      const todoListsWrapper = document.querySelector('.lists');
+      todoListsWrapper.innerHTML = '';
+    }
+  }
+}
```

Comment:
[JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)
- This part of the code with functions could be refactored to a separate module and imported where it is needed.

Reply:
Thank you @jugosack for the comment.